hoodie-client/src/main/java/com/uber/hoodie/func/payload/GenericRecordBufferedIteratorPayload.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.func.payload;
+
+import org.apache.avro.generic.GenericRecord;
+
+public class GenericRecordBufferedIteratorPayload
+    extends AbstractBufferedIteratorPayload<GenericRecord, GenericRecord> {
+
+  public GenericRecordBufferedIteratorPayload(GenericRecord record) {
+    this.inputPayload = record;
```

Comment:
can we move this constructor to the abstract class above.. thats the typical pattern . or did you intend for it to be a Interface originally?

Reply:
yeah, missed to refactor that after other changes, thanks for pointing that out.